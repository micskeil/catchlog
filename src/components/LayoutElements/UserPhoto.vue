<template>
  <div>
    <img class="nav-avatar" :src="updatedPhotoURL" alt="user-avatar" />
  </div>
</template>

<script>
import { db } from "../../firebase";

export default {
  props: {
    uid: {
      type: String,
      required: true,
    },
  },
  data() {
    return {
      photoURL: "",
    };
  },
  computed: {
    updatedPhotoURL() {
      if (this.photoURL === "") {
        return require("../../assets/img/user_1.png");
      } else {
        return this.photoURL;
      }
    },
  },
  methods: {
    getPhotoUrl() {
      const that = this;
      db.collection("users")
        .doc(this.uid)
        .get()
        .then(function(doc) {
          if (doc.exists) {
            that.photoURL = doc.data().photoURL;
          } else {
            // doc.data() will be undefined in this case
            console.log("No such document!");
          }
        })
        .catch(function(error) {
          console.log("Error getting document:", error);
        });
    },
  },
  mounted() {
    if (this.uid === "Guest") {
      console.log("Hello Guest!");
    } else {
      this.getPhotoUrl();
    }
  },
};
</script>

<style scoped>
img {
  width: 100%;
  height: 100%;
  border: 1px solid #c79136;
  box-shadow: 0 0 0 2px #fff, 0 0 0 3px #c79136;
  border-radius: 100%;
}
</style>
