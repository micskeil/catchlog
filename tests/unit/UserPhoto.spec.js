import { shallowMount } from "@vue/test-utils";
import UserPhoto from "../../src/components/LayoutElements/UserPhoto.vue";

describe("UserPhoto", () => {
  it("should give back the defaultPhoto if no uid prop provided ", () => {
    const propsData = { uid: "" };
    const wrapper = shallowMount(UserPhoto, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render to a snapshot when list has items", () => {
    const propsData = { uid: "4FJQKFUuWQdnnWTD7GzGFGvAXZ62" };
    const wrapper = shallowMount(UserPhoto, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });
});
