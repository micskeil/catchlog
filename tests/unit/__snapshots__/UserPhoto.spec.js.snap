// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPhoto should give back the defaultPhoto if no uid prop provided  1`] = `
<div>
  <img
    alt="user-avatar"
    class="nav-avatar"
    src=""
  />
</div>
`;

exports[`UserPhoto should render to a snapshot when list has items 1`] = `
<div>
  <img
    alt="user-avatar"
    class="nav-avatar"
    src=""
  />
</div>
`;
